src/sim/workload.hh

```diff
@@ -103,6 +103,8 @@ class Workload : public SimObject
     virtual Addr getEntry() const = 0;
     virtual ByteOrder byteOrder() const = 0;
     virtual loader::Arch getArch() const = 0;
+    /// Returns the semihosting interface if supported by the current workload.
+    virtual SimObject* getSemihosting() const { return nullptr; }
```

Comment:
I'd suggest not adding this here unless it's used by other architectures.

Reply:
We get the semihost from workload `src/arch/riscv/insts/standard.cc`. https://github.com/gem5/gem5/pull/681/commits/e48114c6581d9493030b155717e4d93ec1600957#diff-e0c1e51647bad6c3927cecd92b966ef5f523489faaf39bc4647b1b712a0e39b8

There are three types of workloads used in RISC-V system, `RiscvBareMetal`, `RiscvLinux` and `RiscvBootloaderKernelWorkload`. I think adding the getSimihosting in generic `Workload` is make sense to me